dipy/align/_public.py

```diff
@@ -0,0 +1,833 @@
+
+    pipeline : sequence, optional
+        Sequence of transforms to use in the gradual fitting of the full
+        affine. Default: (executed from left to right):
+        `[c_of_mass, translation, rigid, affine]`
+
+    starting_affine: 4x4 array, optional
+        Initial guess for the transformation between the spaces.
+        Default: identity.
+
+    metric : str, optional.
+        Currently only supports 'MI' for MutualInformationMetric.
+
+    nbins : int, optional
+        MutualInformationMetric key-word argument: the number of bins to be
+        used for computing the intensity histograms. The default is 32.
+
+    sampling_proportion : None or float in interval (0, 1], optional
+        MutualInformationMetric key-word argument: There are two types of
+        sampling: dense and sparse. Dense sampling uses all voxels for
+        estimating the (joint and marginal) intensity histograms, while
+        sparse sampling uses a subset of them. If `sampling_proportion` is
+        None, then dense sampling is used. If `sampling_proportion` is a
+        floating point value in (0,1] then sparse sampling is used,
+        where `sampling_proportion` specifies the proportion of voxels to
+        be used. The default is None (dense sampling).
+
+    level_iters : sequence, optional
+        AffineRegistration key-word argument: the number of iterations at each
+        scale of the scale space. `level_iters[0]` corresponds to the coarsest
+        scale, `level_iters[-1]` the finest, where n is the length of the
+        sequence. By default, a 3-level scale space with iterations
+        sequence equal to [10000, 1000, 100] will be used.
+
+    sigmas : sequence of floats, optional
+        AffineRegistration key-word argument: custom smoothing parameter to
+        build the scale space (one parameter for each scale). By default,
+        the sequence of sigmas will be [3, 1, 0].
+
+    factors : sequence of floats, optional
+        AffineRegistration key-word argument: custom scale factors to build the
+        scale space (one factor for each scale). By default, the sequence of
+        factors will be [4, 2, 1].
+
+    Returns
+    -------
+    transformed, affine : array with moving data resampled to the static space
+    after computing the affine transformation and the affine 4x4
+    associated with the transformation.
+
+
+    Notes
+    -----
+    Performs a gradual registration between the two inputs, using a pipeline
+    that gradually approximates the final registration. If the final default
+    step (`affine`) is ommitted, the resulting affine may not have all 12
+    degrees of freedom adjusted.
+    """
+    if pipeline is None:
+        pipeline = [c_of_mass, translation, rigid, affine]
+    if level_iters is None:
+        level_iters = [10000, 1000, 100]
+    if sigmas is None:
+        sigmas = [3, 1, 0.0]
+    if factors is None:
+        factors = [4, 2, 1]
```

Comment:
same comment as above for all this block: `factors = factors or [4, 2, 1]`

Reply:
Done in 2cad7da1